expression/errors.go

```diff
@@ -67,6 +67,20 @@ func handleInvalidTimeError(ctx sessionctx.Context, err error) error {
 	return nil
 }
 
+// handleInvalidDurationError reports error or warning depend on the context.
+func handleInvalidDurationError(ctx sessionctx.Context, err error) error {
+	if err == nil || !(types.ErrWrongValue.Equal(err) || types.ErrWrongValueForType.Equal(err) ||
+		types.ErrTruncatedWrongVal.Equal(err)) {
+		return err
+	}
+	sc := ctx.GetSessionVars().StmtCtx
+	err = sc.HandleTruncate(err)
+	if ctx.GetSessionVars().StrictSQLMode && (sc.InInsertStmt || sc.InUpdateStmt || sc.InDeleteStmt) {
```

Comment:
I'm not sure if this line is necessary. The error should be handled by HandleTruncate().